vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -1019,18 +1015,21 @@ import { ItemCache } from '@vaadin/grid/src/vaadin-grid-data-provider-mixin.js';
         }));
       }));
 
-      function _fireClickEvent(event, eventName) {
-        if (grid.$connector.activeItem) {
-          event.itemKey = grid.$connector.activeItem.key;
-          const eventContext = grid.getEventContext(event);
-          // if you have a details-renderer, getEventContext().column is undefined
-          if (eventContext.column) {
-            event.internalColumnId = eventContext.column._flowId;
-          }
-          grid.dispatchEvent(new CustomEvent(eventName,
-            { detail: event }));
+    function _fireClickEvent(event, eventName) {
+      const target = event.target;
+      const eventContext = grid.getEventContext(event);
+      const section = eventContext.section;
+
+      if (eventContext.item && !isFocusable(target) && section !== 'details') {
+        event.itemKey = eventContext.item.key;
+        // if you have a details-renderer, getEventContext().column is undefined
+        if (eventContext.column) {
+          event.internalColumnId = eventContext.column._flowId;
```

Comment:
This check would be unnecessary if you couldn't enter this block if `section === 'details'`?